### PR TITLE
Buying a reward

### PR DESCRIPTION
**Pull request by mdobosz3**
Adding the possibility of ordering rewards by an employee - the value of the reward is deducted from the number of kudos received

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    end
+  end
+
+  private
+
+  def reward
+    Reward.find(params[:reward])
+  end
+
+  def order
```
Comment (ciejj) [inline app/controllers/orders_controller.rb:22]: I think we do not need this method at this point

File: spec/system/order_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+RSpec.describe 'Order check', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  let!(:employee) { create(:employee) }
+  let!(:company_value) { create(:company_value) }
+
+  it 'Buying a reward' do
+    create(:kudo, giver: employee, receiver: employee, company_value: company_value)
```
Comment (ciejj) [inline spec/system/order_spec.rb:14]: Your factory will create `company_value` for you, and can also create another kudo-giver `employee`, so the kudo creation can look like this:

```suggestion
    create(:kudo, receiver: employee)
```

**Review by ciejj — APPROVED** [review]
Nice one! I have left some suggestion for minor improvements

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (mdobosz3) [reply app/controllers/orders_controller.rb]: You're right, I don't need this method

File: config/routes.rb [hunk 3]
```diff
@@ -5,6 +5,7 @@
   devise_for :employees, path: 'employees'
   resources :kudos
   resources :rewards
+  resources :orders
```
[the remark this answers is not in the document]
Comment (mdobosz3) [reply config/routes.rb]: I left only the used paths in orders and rewards

File: spec/system/order_spec.rb (hunk 2 above)
Comment (mdobosz3) [reply spec/system/order_spec.rb]: You're right, I should use the factory if I can :)

File: app/controllers/orders_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+class OrdersController < ApplicationController
+  before_action :authenticate_employee!
+
+  def create
+    if current_employee.kudo_points < reward.price
+      redirect_to rewards_path, notice: "You don't have enough kudos."
+    else
+      order = Order.new(employee: current_employee, reward: reward)
+      order.save
```
Comment (oskarlakner) [inline app/controllers/orders_controller.rb:11]: if you don't need to return validations to a user then I'd just use .create!

File: spec/system/order_spec.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    sign_in employee
+    visit root_path
+    click_link 'Rewards'
+    within('[data-test-id="Kudo_Points"]') do
+      expect(page).to have_content '1'
+    end
+
+    click_link 'Buy'
+    expect(page).to have_content 'Reward was successfully buying.'
+    within('[data-test-id="Kudo_Points"]') do
```
Comment (oskarlakner) [inline spec/system/order_spec.rb:25]: 👏

**Review by oskarlakner — APPROVED** [review]

**Pull request merged**